File: julia/euler/004.jl
function is_palindrome(num::Int)::Bool
 digits::Array{Int, 1} = get_digits(num); len::Int = length(digits)
  for element::Int = [1:ceil(len/2);]
    if digits[element] != digits[len - element + 1] return false end
  end
  return true
end
function palindrome_products()::Int
  palindromes::Array{Int, 1} = []
  for i::Int = [101:999;]
    for j::Int = [i:999;]
      if is_palindrome(i*j) == true
        push!(palindromes,i*j) 
      end
    end
  end
  return maximum(palindromes)
end
function get_digits(num::Int)::Array{Int, 1}
  digits::Array{Int, 1} = []
  while true
    digit::Int = num % 10; push!(digits, digit);
    num::Int = trunc(Int, num / 10); num < 1 && break;
  end
  return digits
end

File: python/euler/004.py
def is_palindrome(num):
    """TODO: Docstring for is_palindrome.
    :num: TODO
    :returns: TODO
    """
    digits = get_digits(num)
    len_ = len(digits)
    for element in range(len_):
        print(digits[element] ,digits[len_ - element - 1])
        if digits[element] != digits[len_ - element - 1]:
            return False
    return True

def palindrome_products():
    """TODO: Docstring for palindrome_products.
    :returns: TODO
    """
    palindromes = []
    for i in reversed(range(100, 999)):
        for j in reversed(range(i, 999)):
            if is_palindrome(i*j):
                palindromes.append(i*j)
    return max(palindromes)

def get_digits(num):
    """TODO: Docstring for get_digits.
    :num: TODO
    :returns: TODO
    """
    digits = []
    while True:
        digit = num % 10
        digits.append(digit)
        num = num // 10
        if num < 1:
            break
    return digits

if __name__ == "__main__":
    pal = palindrome_products()
